src/main/java/dk/kb/image/api/v1/impl/AccessApiServiceImpl.java

```diff
@@ -395,6 +408,77 @@ public StreamingOutput iIPImageRequest(
+     */
+    private boolean isThumbnailIIIF(String identifier, String region, String size, String rotation, String quality, String format) {
+         if (!"full".equals(region)  ||  !rotation.equals("0") || !quality.equals("default")) {        	         	
+        	 log.debug("Fullsize for IIIF request since a custom parameter was defined");             
+             return false;                
+         }
+         else if (size == null) {
+             log.debug("Fullsize for IIIF request since size parameter was not defined");
+             return false;
+         }                           
+         
+            //Only allow size as "w,h" parameter. Etc. "100,100". Value is default 'max' when requesting the full image.
+         String[] tokens = size.substring(1).split(","); //First character is !
+         if (tokens.length != 2) {
+             log.debug("Fullsize for IIIF request since size parameter not width,hight");                          
+             return false;
+         }
+         int width;
+         int height;         
+         try {
+             width=Integer.parseInt(tokens[0].trim());
+             height=Integer.parseInt(tokens[1].trim());
+         }
+         catch(Exception e) {
+             log.debug("Fullsize for IIIF request since size parameter could not be parsed as (width,height) integers:"+size);
+             return false;
+         }
+                  
+        if ( width > ServiceConfig.getConfig().getInteger("thumbnail.maxWidth") || height > ServiceConfig.getConfig().getInteger("thumbnail.maxHeight")){         
+             log.debug("Fullsize for IIIF request since size parameter was over thumbnail size");             
+             return false;
+         }
+         
+        return true;
+    }
+
+
+    
+    /* Is request classified as a thumbnail or fullsize for IIP requests.
+     * 
+     * This implementation is very conservative and will determine thumbnail also if most non size-parameters are defined.
+     * It is better to be conservative and later loosen up than giving too much control over thumbnail extraction.
+     * 
+     * Will be full size if most other parameters than FIF and CVT is defined. Also WID and HEI must be below a defined limit in the configuration or it will also be fullsize.   
+     */
+    private boolean isThumbnailIIP( String FIF, Long WID, Long HEI, List<Float> RGN, Integer QLT, Float CNT, String ROT, Float GAM, String CMP, String PFL, String CTW, Boolean INV, String COL,
+            List<Integer> JTL, List<Integer> PTL, String CVT) {
+    	
+    	//FIF and CVT allowed. WID and HEI checked below
+        if (  (RGN != null && RGN.size() !=0) || QLT != null || CNT != null || ROT != null || GAM != null || CMP != null || PFL != null || INV != null || COL != null  || (JTL != null && JTL.size() >0 )) {        	
```

Comment:
Our own OpenAPI has `ROT=0` as default, so that should be accepted.

Also, `PTL` is not checked and can be used to reconstruct the full image.

Reply:
ROT=0 now allowed: http://devel11:10001/ds-image/v1/IIP/?FIF=/DAMJP2/DAM/Samlingsbilleder/0000/750/978/DNF_1955-00298_00092&HEI=150&ROT=0&WID=150&CVT=jpeg

PTL added. Left by mistake.